lib/services/account-api-service.js

```diff
@@ -0,0 +1,193 @@
+        var authEnabled = _(endpoints).filter(function(endpoint) {
+            return endpoint.authEnabled === true;
+        }).compact().value();
+
+        // Only initialize the service if an endpoint has enabled authn
+        if(_.isEmpty(authEnabled)) {
+            return;
+        }
+
+        return Promise.join( self.listUsers(), waterline.roles.find({}), function(users, roles) {
+            var promises = _.map(users, function(user) {
+                    return acl.addUserRoles(user.username, user.role);
+                });
+            var rolePromises = _.map(roles, function(roles) {
+                    return acl.addRoleParents(roles.role, roles.privileges);
+                });
+            Array.prototype.push.apply(promises, rolePromises);
+            return Promise.all(promises);
+        }).then(function() {
+            // Enforce a subset of hardcoded privileges:
+            return Promise.all([
+                // 2.0 API's fixed privileges
+                acl.addRoleParents('Administrator', ['Read', 'Write']),
+                acl.addRoleParents('ReadOnly', ['Read']),
+
+                // Redfish 1.0 fixed privileges
+                acl.addRoleParents('Administrator', ['Login','ConfigureManager','ConfigureUsers']),
+                acl.addRoleParents('Administrator', ['ConfigureComponents','ConfigureSelf']),
+                acl.addRoleParents('Operator', ['Login','ConfigureComponents','ConfigureSelf']),
+                acl.addRoleParents('ReadOnly', ['Login', 'ConfigureSelf'])
+            ]);
+        });
+    };
+
+    AccountApiService.prototype.aclMethod = function(name) {
+        var args = Array.prototype.slice.call(arguments, 1);
+        return this.acl[name].apply(this.acl, args);
+    };
+
+    AccountApiService.prototype.listUsers = function() {
+        var self = this;
+        return Promise.map(waterline.localusers.find({}), function(user) {
+            return Promise.props({
+                id: user.id,
+                username: user.username,
+                role: user.role,
+                privileges: self.aclMethod('_rolesParents', [user.role])
+            });
+        });
+    };
+
+    AccountApiService.prototype.getUserByName = function(username) {
+        var self = this;
+        return Promise.try(function() {
+            assert.string(username);
+            return waterline.localusers.findOne({username: username});
+        })
+        .then(function(user) {
+            if(!user) {
+                throw new Errors.NotFoundError();
+            }
+            return Promise.props({
+                id: user.id,
+                username: user.username,
+                role: user.role,
+                privileges: self.aclMethod('_rolesParents', [user.role])
+            });
+        });
+    };
+
+    AccountApiService.prototype.createUser = function(user) {
+        var self = this;
+        return Promise.try(function() {
+            assert.string(user.username);
```

Comment:
2.0 and redfish both use swagger validation.  Your feedback below about adding negative unittests would make these assertions a requirement.  So while this code is unnecessary for production or functional use, it would strictly be required to satisfy your need to test for conditions that cannot occur.